src/code/z_snap.c

```diff
@@ -119,42 +147,49 @@ s32 func_8013A530(PlayState* play, Actor* actor, s32 flag, Vec3f* pos, Vec3s* ro
     s32 ret = 0;
     s32 bgId;
 
+    // Check distance
     distance = OLib_Vec3fDist(pos, &camera->eye);
     if ((distance < distanceMin) || (distanceMax < distance)) {
-        func_8013A41C(0x3F);
-        ret = 0x3F;
+        Snap_SetFlag(PICTOGRAPH_BAD_DISTANCE);
+        ret = PICTOGRAPH_BAD_DISTANCE;
     }
 
-    x = func_8013A504(Camera_GetCamDirPitch(camera) + rot->x);
-    y = func_8013A504(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
-    if ((0 < angleError) && ((angleError < x) || (angleError < y))) {
-        func_8013A41C(0x3E);
-        ret |= 0x3E;
+    // Check rot is facing camera?
+    x = Snap_AbsS(Camera_GetCamDirPitch(camera) + rot->x);
+    y = Snap_AbsS(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
+    if ((0 < angleRange) && ((angleRange < x) || (angleRange < y))) {
+        Snap_SetFlag(PICTOGRAPH_BAD_ANGLE);
+        ret |= PICTOGRAPH_BAD_ANGLE;
     }
 
-    Actor_GetProjectedPos(play, pos, &screenSpace, &distance);
-    x = (s16)(screenSpace.x * distance * 160.0f + 160.0f) - 85;
-    y = (s16)(screenSpace.y * distance * -120.0f + 120.0f) - 67;
-    if ((x < 0) || (150 < x) || (y < 0) || (105 < y)) {
-        func_8013A41C(0x3D);
-        ret |= 0x3D;
+    // Check in viewfinder
+    Actor_GetProjectedPos(play, pos, &projectedPos, &distance);
+    //! TODO: relate these numbers to the pictograph box interface / capture
+    x = (s16)(projectedPos.x * distance * 160.0f + 160.0f) - 85;
+    y = (s16)(projectedPos.y * distance * -120.0f + 120.0f) - 67;
+    if ((x < 0) || (x > 150) || (y < 0) || (y > 105)) {
```

Comment:
```suggestion
    #define PICTO_SCREEN_WIDTH 150
    #define PICTO_SCREEN_HEIGHT 105
    
    Actor_GetProjectedPos(play, pos, &projectedPos, &distance);
    
   // Get the screen coordinates, then shift the origin to the top left corner of the smaller interior pictoscreen
    x = (s16)(projectedPos.x * distance * (SCREEN_WIDTH / 2) + (SCREEN_WIDTH / 2)) - ((SCREEN_WIDTH 
 - PICTO_SCREEN_WIDTH) / 2);
    y = (s16)(projectedPos.y * distance * -(SCREEN_HEIGHT / 2) +(SCREEN_HEIGHT / 2)) - ((SCREEN_HEIGHT 
 - PICTO_SCREEN_HEIGHT) / 2);
 
    // checks if the coordinates are within the sub-screen of the picto-photo
    if ((x < 0) || (x > PICTO_SCREEN_WIDTH) || (y < 0) || (y > PICTO_SCREEN_HEIGHT)) {
```
Yeah, your assumption was right. This gets the screen coordinates, shifts the origin of the coordinates to the top-left corner of that smaller rectangle screen for the pictograph, than checks if it's within that smaller screen.

An alternative define would be to `#define PICTO_SCREEN_CUTOFF 85` define the dimensions of the screen that's cutoff, instead of the dimensions of the interior picto size, but that seems less intuitive

Reply:
So I wonder if `PICTOGRAPH_NOT_IN_VIEW` should be `PICTOGRAPH_NOT_IN_PICTO_SCREEN` or `PICTOGRAPH_NOT_IN_SUB_SCREEN` or something?

Reply:
`NOT_IN_PICTO_CAPTURE_AREA`?

Reply:
And it's more that it's the top-left corner of the capture region for the initial shifting. I'll think about it myself, see what I come up with.

Reply:
Would `RESOLUTION` be a reasonable description, or is that unrelated to the actual screen coordinates?

Reply:
Resolution could work for the actual picto frame. Would also prefer not name it screen width or height. Capture area or some other suggestion like that is better. I don't think it needs to hold up merging because you stated documentation is not totally complete.